xenserver/logrotate: only restart services which are running

In particular this prevents:
  ovs-appctl: cannot read pidfile "/var/run/openvswitch/ovs-brcompatd.pid" (No such file or directory)
when ovs-brcompatd is disabled.

Signed-off-by: Ian Campbell <ian.campbell@citrix.com>

diff -r 263448afccaa xenserver/etc_logrotate.d_openvswitch
--- a/xenserver/etc_logrotate.d_openvswitch	Mon Mar 29 10:20:16 2010 +0100
+++ b/xenserver/etc_logrotate.d_openvswitch	Mon Mar 29 13:53:32 2010 +0100
@@ -10,8 +10,8 @@
 	missingok
 	postrotate
 	# Tell Open vSwitch daemons to reopen their log files
-	/usr/bin/ovs-appctl -t ovs-vswitchd vlog/reopen
-	/usr/bin/ovs-appctl -t ovs-brcompatd vlog/reopen
-	/usr/bin/ovs-appctl -t ovsdb-server vlog/reopen
+	[ ! -f /var/run/openvswitch/ovs-vswitchd.pid ] || /usr/bin/ovs-appctl -t ovs-vswitchd vlog/reopen
+	[ ! -f /var/run/openvswitch/ovs-brcompatd.pid ] || /usr/bin/ovs-appctl -t ovs-brcompatd vlog/reopen
+	[ ! -f /var/run/openvswitch/ovsdb-server.pid ] || /usr/bin/ovs-appctl -t ovsdb-server vlog/reopen
 	endscript
 }
